### ISSUE TITLE
`AddedToken` 's argument are ignored when called in `add_tokens` 's method of slow tokenizers

### ISSUE THREAD

**Issue by SaulLu** [huggingface/transformers#20734]
### System Info

- `transformers` version: 4.25.1
- Platform: Linux-5.10.133+-x86_64-with-glibc2.27
- Python version: 3.8.16
- Huggingface_hub version: 0.11.1
- PyTorch version (GPU?): 1.13.0+cu116 (False)
- Tensorflow version (GPU?): 2.9.2 (False)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using GPU in script?: No
- Using distributed or parallel set-up in script?: No

### Who can help?

@ArthurZucker 

### Information

- [ ] The official example scripts
- [X] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [X] My own task or dataset (give details below)

### Reproduction

The explanations of the bug and its reproduction are contained in the following google colab:  https://colab.research.google.com/drive/19SS6Tzlgo0vntFtM6ZsCYq8BNZ5Dy1cS?usp=sharing

### Expected behavior

I would expect the fast and slow tokenizers to treat the `AddedToken`'s arguments in the same way.

I think the loss of information for the slow tokenizer occurs at this line:  https://github.com/huggingface/transformers/blob/a413c725d40027134f28f92974ad7d61751f5640/src/transformers/tokenization_utils.py#L411

**Comment by ArthurZucker** [comment]
I have not dropped this, its still on my TODO list. There are a lot of linked issues!

**Comment by ArthurZucker** [comment]
(The update is gonna take longer as I am refactoring the tokenizers)

**Comment by superRookie007** [comment]
I had the same issue. It seems that `AddedToken` is implemented [here](https://github.com/huggingface/transformers/blob/e42587f596181396e1c4b63660abf0c736b10dae/src/transformers/tokenization_utils_base.py#L80), if the package `tokenizers` is not available. However, the python implementation using dataclasses does not behave the same way as the rust implementation in `tokenizers`.
